opm/output/eclipse/Summary.cpp

```diff
@@ -2021,13 +2024,61 @@ quantity group_efficiency_factor(const fn_args& args)
         return zero;
     }
 
-    const auto gefac =
-        args.schedule[args.sim_step].groups(args.group_name)
-        .getGroupEfficiencyFactor();
+    const auto sched = args.schedule[args.sim_step];
+    const auto gefac = sched.groups.has(args.group_name)
+                        ? sched.groups(args.group_name).getGroupEfficiencyFactor()
+                        : 1.0;
 
     return { gefac, measure::identity };
 }
 
+quantity gas_consumption_rate(const fn_args& args)
+{
+    auto gcr_recursive =
+        [&args](auto self, const std::string& gname) -> double
+        {
+            double consumption_rate = 0.0;
+            const auto& schedule = args.schedule[args.sim_step];
+            const auto& gconsump = schedule.gconsump();
+            if (schedule.groups.has(gname)) {
+                for (const auto& child_gname : schedule.groups(gname).groups()) {
+                    const double gefac = schedule.groups(child_gname).getGroupEfficiencyFactor();
+                    consumption_rate += (gefac * self(self, child_gname));
+                }
+            }
+            if (gconsump.has(gname)) {
+                consumption_rate += gconsump.get(gname, args.st).consumption_rate;
+            }
+            return consumption_rate;
+        };
+
+    return { gcr_recursive(gcr_recursive, args.group_name), measure::gas_surface_rate };
```

Comment:
I think we can fuse the implementations of the consumption and import rates if we make a helper function in terms of a pointer to member data.  For instance,
```C++
double gconsump_rate(const std::string& gname,
                     const Opm::ScheduleState& schedule,
                     const Opm::SummaryState& st,
                     double Opm::GConSump::GCONSUMPGroupProp::* rate)
{
    double tot_rate = 0.0;
    if (schedule.groups.has(gname)) {
        for (const auto& child : schedule.groups(gname).groups()) {
            const auto efac = schedule.groups(child).getGroupEfficiencyFactor();
            tot_rate += efac * gconsump_rate(child, schedule, st, rate);
        }
    }

    if (const auto& gconsump = schedule.gconsump(); gconsump.has(gname)) {
        tot_rate += gconsump.get(gname, st).*rate;
    }

    return tot_rate;
}
```
Then I think you could implement `gas_consumption_rate()` akin to
```C++
quantity gas_consumption_rate(const fn_args& args)
{
    return {
        gconsump_rate(args.group_name,
                      args.schedule[args.sim_step], args.st,
                      &GConSump::GCONSUMPGroupProp::consumption_rate),
        measure::gas_surface_rate
    };
}
```
and `gas_import_rate()` would be something along the lines of
```C++
quantity gas_consumption_rate(const fn_args& args)
{
    return {
        gconsump_rate(args.group_name,
                      args.schedule[args.sim_step], args.st,
                      &GConSump::GCONSUMPGroupProp::import_rate),
        measure::gas_surface_rate
    };
}
```

Reply:
Nice! I'll try this.